pyphs/core/tools/_subsinverse.py

```diff
@@ -55,13 +55,11 @@ def subsinverse_vector(expr, subs, symbols):
         The resulting expression.
     """
     # recast as list
-    if not isinstance(expr, types.vector_types[0]):
-        expr = types.vector_types[0](expr)
-
+    temp = list(expr)
```

Comment:
I can see the following pattern repeated several times : list creation, item assignment, conversion to tuple.
May be worth adding a __setitem__ attribute to PHSVector:
```
class PHSVector(sy.Tuple):
    def __setitem__(self, index, value): 
        # Necessary access to _args, as args is read-only property
        self._args = [v if (i != index) else value for i, v in enumerate(self._args)]
```